### PR TITLE
Make S3 SelectObjectContentInput ScanRange members pointers

### PR DESCRIPTION
**Pull request by cad**
### Problem

**TLDR: Int64 members `Start` and `End` of `ScanRange` for the  `SelectObjectContentInput` are omitted when serialized if they are 0.**

When calling SelectObjectContent, you can specify a ScanRange to only request
records between start and end positions.

For example we want to process only the records starting from the 0th byte to the 100th byte we can have a Go code like this:

```go
cfg, err := config.LoadDefaultConfig(context.TODO(),
    config.WithClientLogMode(aws.LogRequestWithBody|aws.LogResponse), // Enable debug logging API calls
)
if err != nil {
    panic(err)
}

// Create S3 client
client := s3.NewFromConfig(cfg)


// Send a SelectObjectContent request
if _, err := s3Client.SelectObjectContent(ctx, &s3.SelectObjectContentInput{
    Bucket: aws.String("bucket"),
    Key:    aws.String("key"),
    Expression: aws.String("SELECT * FROM S3Object"),
    ExpressionType: s3.ExpressionTypeSql,
    InputSerialization: &s3.InputSerialization{
        JSON: &s3.JSONInput{
            Type: s3.JSONTypeDocument,
        },
    },
    OutputSerialization: &s3.OutputSerialization{
        JSON: &s3.JSONOutput{},
    },
    ScanRange: &s3.ScanRange{ // Expected to get records starting from the 0th byte to the 100th byte
        Start: 0, 
        End:   100,
    },
}); err != nil {
    panic(err)
}
```

Expected API call should have a ScanRange section that looks like this:

```xml
...
<ScanRange>
    <End>100</End>
    <Start>0</Start>
</ScanRange>
...
```

But when checked the log output, it looks like this:

```xml
...
<ScanRange>
    <End>100</End>
</ScanRange>
...
```

Which meant, *"process only the records within the last 100 bytes of the file"*
according to the
[docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html#AmazonS3-SelectObjectContent-request-ScanRange).

**So there is no way to specify the start position as 0 and end position as an arbitrary number, of the ScanRange at the current state of the SDK.**

#### Hypothesis

The client just omits the `Start` or `End` member of the `ScanRange` shape when it is
equal to 0. Since it is an unboxed type and its empty value is 0 it is not serialized to the request.

### Solution

Added a customization that adds `clientOptional` trait to the `Start` and `End`
members of the S3 `ScanRange` shape. And regenerate the client code to make
these members optional.

Since this will break any existing application that specifies `ScanRange` with `Start` and `End` fields set, an announcement is added to the changelog regarding the breaking change.

### CODE REVIEW

**Comment by lucix-aws** [comment]
This is actually part of a larger-scale issue with default value serialization. See https://github.com/aws/aws-sdk-go-v2/issues/2162.

We can't take this PR in isolation due to the breaking change, but we're investigating the issue at large to try to determine a more comprehensive way forward.
